theatre/studio/src/panels/SequenceEditorPanel/RightOverlay/Playhead.tsx

```diff
@@ -249,6 +251,12 @@ const Playhead: React.FC<{layoutP: Pointer<SequenceEditorPanelLayout>}> = ({
   // hide the frame stamp when seeking
   useLockFrameStampPosition(useVal(layoutP.seeker.isSeeking) || isDragging, -1)
 
+  const [contextMenu] = useMarkerContextMenu(
+    thumbNode,
+    val(layoutP),
```

Comment:
Multiple issues here.

1 - `val()` can _only_ be called inside a `prism()`. Otherwise nothing will track the changes to the underlying derivation.
You're calling it inside a react component. Either use `useVal()` or `usePrism()` here.

Reply:
2 - There would still be a problem if you corrected this to to `usePrism(() => val(layoutP))`.

It would work, but it would be inefficient. Any changes to any part of the layout that is unrelated to markers would still cause the `Playhead` to re-render.

To fix this, you need to point inside `layoutP`, for example: `layoutP.sheet`, which would update only when the user picks a new sheet.

Reply:
3 - We don't even need to know the `val(layoutP.sheet)` or `val(layoutP.sheet).getSequence().position` unless the user actually R-clicks on the playhead.

So better to move these calculations to the `items()` callback in `useContextMenu(node, {items})`.

There, you _can_ use `val()` which won't be reactive, but you won't need reactivity there.